lib/fizzy/parser_expr.cpp

```diff
@@ -861,11 +861,12 @@ parser_result<Code> parse_expr(const uint8_t* pos, const uint8_t* end, FuncIdx f
 
             uint32_t offset;
             std::tie(offset, pos) = leb128u_decode<uint32_t>(pos, end);
-            push(code.immediates, offset);
+            code.instructions.push_back(opcode);
+            push(code.instructions, offset);
 
             if (!module.has_memory())
                 throw validation_error{"memory instructions require imported or defined memory"};
-            break;
+            continue;
```

Comment:
Is this continue here jumping to the while body? I did not know you can jump out from within switches with it.

Reply:
This is a bit hackish, but also "minimal" change. The `continue` only affects loops but `break` both loops and switches.